cmd/k8s-operator/svc-for-pg.go

```diff
@@ -0,0 +1,899 @@
+	// 'if or.OperatorID === r.operatorID || or.ingressUID == r.ingressUID'.
+	ix := slices.IndexFunc(o.OwnerRefs, func(or OwnerRef) bool {
+		return or.OperatorID == r.operatorID
+	})
+	if ix == -1 {
+		return false, nil
+	}
+	if len(o.OwnerRefs) == 1 {
+		logger.Infof("Deleting VIPService %q", name)
+		return false, r.tsClient.DeleteVIPService(ctx, name)
+	}
+	o.OwnerRefs = slices.Delete(o.OwnerRefs, ix, ix+1)
+	logger.Infof("Deleting VIPService %q", name)
+	json, err := json.Marshal(o)
+	if err != nil {
+		return false, fmt.Errorf("error marshalling updated VIPService owner reference: %w", err)
+	}
+	svc.Annotations[ownerAnnotation] = string(json)
+	return true, r.tsClient.CreateOrUpdateVIPService(ctx, svc)
+}
+
+func (a *HAServiceReconciler) backendRoutesSetup(ctx context.Context, serviceName, replicaName, pgName string, wantsCfg *ingressservices.Config, logger *zap.SugaredLogger) (bool, error) {
+	logger.Debugf("checking backend routes for service '%s'", serviceName)
+	pod := &corev1.Pod{}
+	err := a.Get(ctx, client.ObjectKey{Namespace: a.tsNamespace, Name: replicaName}, pod)
+	if apierrors.IsNotFound(err) {
+		logger.Debugf("Pod %q not found", replicaName)
+		return false, nil
+	}
+	if err != nil {
+		return false, fmt.Errorf("failed to get Pod: %w", err)
+	}
+	secret := &corev1.Secret{}
+	err = a.Get(ctx, client.ObjectKey{Namespace: a.tsNamespace, Name: replicaName}, secret)
+	if apierrors.IsNotFound(err) {
+		logger.Debugf("Secret %q not found", replicaName)
+		return false, nil
+	}
+	if err != nil {
+		return false, fmt.Errorf("failed to get Secret: %w", err)
+	}
+	if len(secret.Data) == 0 || secret.Data[ingressservices.IngressConfigKey] == nil {
+		return false, nil
+	}
+	gotCfgB := secret.Data[ingressservices.IngressConfigKey]
+	var gotCfgs ingressservices.Status
+	if err := json.Unmarshal(gotCfgB, &gotCfgs); err != nil {
+		return false, fmt.Errorf("error unmarshalling ingress config: %w", err)
+	}
+	statusUpToDate, err := isCurrentStatus(gotCfgs, pod, logger)
+	if err != nil {
+		return false, fmt.Errorf("error checking ingress config status: %w", err)
+	}
+	if !statusUpToDate {
+		logger.Debugf("Pod %q is not ready to advertise VIPService", pod.Name)
+		return false, nil
+	}
+	if !reflect.DeepEqual(gotCfgs.Configs.GetConfig(serviceName), wantsCfg) {
+		wantj, err := json.Marshal(wantsCfg)
+		if err != nil {
+			return false, fmt.Errorf("error marshalling ingress config: %w", err)
+		}
+
+		logger.Debugf("Pod %q has config %q, but wants %q", pod.Name, string(gotCfgB), wantj)
+		return false, nil
+	}
+	return true, nil
+}
+
+func isCurrentStatus(gotCfgs ingressservices.Status, pod *corev1.Pod, logger *zap.SugaredLogger) (bool, error) {
+	ips := pod.Status.PodIPs
+	if len(ips) == 0 {
+		logger.Debugf("Pod %q does not yet have IPs, unable to determine if status is up to date", pod.Name)
+		return false, nil
+	}
+
+	if len(ips) > 2 {
+		return false, fmt.Errorf("POD_IPs can contain at most 2 IPs, got %d (%v)", len(ips), ips)
```

Comment:
I think in this case we don't want to return an error that would cause a reconcile? I don't think the number of Pod IPs would have changed on reconcile